src/tasks/buildAndUpload.ts

```diff
@@ -65,6 +66,7 @@ export function buildAndUpload({
   deleteOldPins?: boolean;
   composeFileName: string;
   dir: string;
+  hardwareArchitecture: Architecture;
```

Comment:
Why pass this as an argument when you could compute it directly in the body of the function? Adds unnecessary code duplication

Reply:
How it was an async function, I had to obtain that value outside of the build function so this function was sync, but after changing the getArchitecture() to sync function, it's no a problem to include into the body of the build function.
Fixed